src/Beberlei/Metrics/Exception/NotSupportedException.php

```diff
@@ -0,0 +1,18 @@
+<?php
+namespace Beberlei\Metrics\Exception;
+
+class NotSupportedException extends \DomainException
+{
+    public function __construct()
+    {
+        $trace = debug_backtrace();
```

Comment:
``debug_backtrace`` has a non-negligeable performance impact (especially when retrieving a full backtrace in case of deep callgraph). You should rather pass the necessary arguments